Nef_S2/include/CGAL/Nef_S2/Sphere_direction.h

```diff
@@ -57,8 +57,8 @@ the great circle through $p$ and $q$ (oriented such that the segment
 $pq$ is the shorter one of the two possible ones. \precond $p$ and $q$
 are not opposite on $S_2$.}*/
 { CGAL_assertion(p!=q.opposite());
-  Point_3 p1(0,0,0), p4 = CGAL::ORIGIN + ((Base*) this)->orthogonal_vector();
-  if ( CGAL::orientation(p1,p,q,p4) != CGAL::POSITIVE )
+  Point_3 p4 = CGAL::ORIGIN + ((Base*) this)->orthogonal_vector();
+  if ( CGAL::orientation(CGAL::ORIGIN,p,q,p4) != CGAL::POSITIVE )
```

Comment:
If I remember well, in PR #6399 you decided not to add the optimized overload to the global function. See https://github.com/CGAL/cgal/pull/6399#issuecomment-1071565003.

Reply:
That's well spotted! I also think this change is orthogonal to the allocation of a vector in K3_tree.

Reply:
Seems the function is not called. I had to fix exacty that in line 105 below.

Reply:
The file contains some more places where we should replace `Point_3(0,0,0)`.  Is it ok if we deal with this in this PR?

Reply:
If the commits are atomic, and the PR title/description matches the change scope, this workflow works for me. ;)